Source/Game/Game/Editor/AssetBrowser.cpp

```diff
@@ -0,0 +1,490 @@
+#include "AssetBrowser.h"
+
+#include <Base/Util/DebugHandler.h>
+#include <Game/Util/ImGui/FakeScrollingArea.h>
+
+namespace Editor
+{
+    AssetBrowser::AssetBrowser()
+        : BaseEditor(GetName(), true),
+          _tree(nullptr), _currentNode(nullptr)
+    {
+        fs::path relativeDataPath = fs::absolute("Data");
+        if (!fs::is_directory(relativeDataPath))
+        {
+            DebugHandler::PrintError("Failed to find Data/ folder from ({0})", relativeDataPath.string());
+            return;
+        }
+
+        _topPath = relativeDataPath;
+        _tree = new TreeNode(_topPath);
+        ProcessTree(_tree);
+        _currentNode = _tree;
+    }
+
+    void AssetBrowser::DrawImGui()
+    {
+        _gameRenderer = ServiceLocator::GetGameRenderer();
+        _renderer = _gameRenderer->GetRenderer();
+
+        if (_averageFontWidth < 0.f)
+            CalculateAverageFontWidth();
+
+        if (ImGui::Begin(GetName()))
+        {
+
+            if (!_defaultImageHandle)
+            {
+                Renderer::TextureDesc textureDesc;
+                textureDesc.path = "Data/Texture/interface/icons/ability_gouge.dds"; // need to have a default image ??
+                auto textureID = _renderer->LoadTexture(textureDesc);
+                _defaultImageHandle = _renderer->GetImguiImageHandle(textureID);
+            }
+
+            ImGui::Text("Actual data folder : %s", _topPath.string().c_str());
+            if (_currentNode)
+            {
+                if (IsHorizontal())
+                    ImGui::SameLine();
+
+                ImGui::Text("%s", _currentNode->ProcessPath(_topPath.string()).c_str());
+            }
+            ImGui::Separator();
+
+            // docked from bottom / top / default
+            if (IsHorizontal())
+            {
+                f32 maxHeight = ImGui::GetContentRegionAvail().y * 0.9f;
+                ImGui::Columns(2, "AssetBrowserColumn", true);
+                ImGui::SetColumnWidth(0, ImGui::GetWindowWidth() * 0.25f);
```

Comment:
Doesn't this override the divider so we can't resize the two parts of the panel?